src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php

```diff
@@ -65,6 +65,23 @@ private function hasDefaultValue(\ReflectionParameter $parameter)
     }
 
     /**
+     * Returns if the argument is allowed to be null but is still mandatory.
+     *
+     * @param \ReflectionParameter $parameter
+     *
+     * @return bool
+     */
+    private function isNullable(\ReflectionParameter $parameter)
+    {
+        if (PHP_VERSION_ID >= 70000) {
```

Comment:
I don't think that's a hack as it actually checks that the feature you need is available (it's imo less hacky than depending on a version). For the same reason, I think we should update the examples you're pointing as it could create unexpected behaviors.

But I'm not using hvvm so it won't impact me anyway, I just think it would be weird if it doesn't work sometimes when it should. 
Moreover the solution to this is not complicated: `class_exists($parameter, 'getType')`.

Maybe we could ask someone of the hvvm team whether it's safe to depend on this constant for this feature?